client/eth/connection_pool.go

```diff
@@ -36,6 +36,17 @@ func NewConnectionPoolImpl(cfg ConnectionPoolConfig, logger log.Logger) (Connect
 		cfg.HealthCheckInterval = defaultHealthCheckInterval
 	}
 
+	// The LRU cache needs at least one URL provided for both HTTP and WS.
+	hasEthHttpUrls := len(cfg.EthHTTPURLs) > 0
+	if !hasEthHttpUrls {
+		return nil, fmt.Errorf("ConnectionPool: missing URL for HTTP clients")
+	}
+
+	hasEthWSUrls := len(cfg.EthWSURLs) > 0
+	if !hasEthWSUrls {
```

Comment:
do we need to enforce that any application using this conn pool must have a WS rpc? 1) a lot of apps don't always need or have access to WS. 2) if the application makes a call to a WS function (like `SubscribeNewHeads`), we can fail there imo.